packages/core/src/codewhispererChat/controllers/chat/messenger/messenger.ts

```diff
@@ -88,6 +89,21 @@ export class Messenger {
             )
         )
     }
+
+    public async countTotalNumberOfCodeBlocks(message: string): Promise<number> {
+        if (message === undefined) {
+            return 0
+        }
+
+        // To Convert Markdown text to HTML using marked library
+        const html = await marked(message)
+
+        // Count the number of <pre> tags in the HTML to find the total number of code blocks
+        const totalNumberOfCodeBlocks = (html.match(/<pre>/g) || []).length
```

Comment:
This is not specific enough - it could match unrelated `<pre>` tags.

Reply:
My suggestion would be [jsdom](https://www.npmjs.com/package/jsdom) and the method [querySelectorAll](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) with a query of `pre > code` to find all code blocks